cgi-bin/LJ/Sysban.pm

```diff
@@ -209,6 +209,12 @@ sub sysban_check {
         return 0 unless scalar @domains >= 2;
         my $domain = "$domains[-2].$domains[-1]";
         return 1 if $check->('email_domain', $domain);
+        # also check for three-element domain bans
+        if ( defined $domains[-3] ) {
```

Comment:
This should be a length check as the previous check. `scalar @domains >= 3` kind of thing.